pkg/httpclient/client.go

```diff
@@ -14,58 +14,85 @@ import (
 
 // A Client is an HTTP client.
 type Client struct {
-	acsToken   string
 	baseURL    string
-	timeout    time.Duration
 	baseClient *http.Client
-	logger     *logrus.Logger
+	ctx        context.Context
 }
 
 // Option is a functional option for an HTTP client.
-type Option func(*Client)
+type Option func(context.Context) context.Context
```

Comment:
Why are you making this change? Why is it necessary for `context` to keep values that are later used to modify HTTP requests behavior (i.e. ACS token).

Reply:
Yes, my main point was about using the context object for options sharing. My perception of context is that it should enhance information going through request, but not make request behaviour dependent on the context.

Reply:
I can agree with that, introducing a dedicated `Options` struct should be cleaner.